src/Nancy/HttpLink.cs

```diff
@@ -0,0 +1,504 @@
+namespace Nancy
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using Nancy.Responses.Negotiation;
+
+    /// <summary>
+    /// Represents one of possibly many RFC 5899 HTTP Links contained in a <see cref="HttpLinkHeader"/>.
+    /// </summary>
+    public class HttpLink : Dictionary<string, object>, IEquatable<HttpLink>

```

Comment:
This class has _a lot_ of constructor overloads - 11 if I am counting correctly. All off these can't be needed?